### PR TITLE
Add an understandable error message to vara-cs when no paper-config exits

### PR DESCRIPTION
**Pull request by Sinerum**
Add an error message explaining how to create a paper-config when no paper-config was found by vara-cs
closes se-passau/VaRA#719

### CODE REVIEW

**Review by boehmseb — CHANGES_REQUESTED** [review]
Can you please make this a decorator so that we can use this like this:
```python
@configuration_lookup_error_handler
def __casestudy_exec_command(...):
    ...
```

File: varats/varats/tools/tool_util.py [hunk 1]
```diff
@@ -64,3 +65,19 @@ def get_research_tool(
 def get_supported_research_tool_names() -> tp.List[str]:
     """Returns a list of all supported research tools."""
     return ["phasar", "vara"]
+
+
+def configuration_lookup_error_handler(func) -> tp.Any:
+    """Wrapper for drivers to catche internal Exceptions and provide a helpful
+    message to the user."""
+
+    def wrapper_configuration_lookup_error_handler(*args, **kwargs):
```
Comment (boehmseb) [inline varats/varats/tools/tool_util.py]:
```suggestion
    @wraps(func)
    def wrapper_configuration_lookup_error_handler(*args, **kwargs):
```

See here for an explanation: https://stackoverflow.com/questions/308999/what-does-functools-wraps-do

**Review by boehmseb — CHANGES_REQUESTED** [review]
Looks good now apart from the two missing type annotations and the small detail I annotated.

**Review by vulder — APPROVED** [review]
Looks good from my side, please clean up the open issues (mypy).

File: varats/varats/tools/tool_util.py [hunk 2]
```diff
@@ -64,3 +66,22 @@ def get_research_tool(
 def get_supported_research_tool_names() -> tp.List[str]:
     """Returns a list of all supported research tools."""
     return ["phasar", "vara"]
+
+
+def configuration_lookup_error_handler(func: tp.Any) -> tp.Any:
+    """Wrapper for drivers to catche internal Exceptions and provide a helpful
```
Comment (boehmseb) [inline varats/varats/tools/tool_util.py]:
```suggestion
    """Wrapper for drivers to catch internal Exceptions and provide a helpful
```

File: varats/varats/tools/tool_util.py [hunk 3]
```diff
@@ -64,3 +66,22 @@ def get_research_tool(
...
+    message to the user."""
+
+    @wraps(func)
+    def wrapper_configuration_lookup_error_handler(
+        *args: tp.Any, **kwargs: tp.Any
+    ) -> tp.Any:
```
Comment (boehmseb) [inline varats/varats/tools/tool_util.py]: I think, we should restrict the return type to be a function:
```suggestion
    ) -> tp.Callable[..., tp.Any]:
```

**Review by boehmseb — CHANGES_REQUESTED** [review]

File: varats/varats/tools/tool_util.py [hunk 4]
```diff
@@ -64,3 +66,22 @@ def get_research_tool(
...
+
+
+def configuration_lookup_error_handler(func: tp.Any) -> tp.Any:
+    """Wrapper for drivers to catch internal Exceptions and provide a helpful
+    message to the user."""
+
+    @wraps(func)
+    def wrapper_configuration_lookup_error_handler(
+        *args: tp.Any, **kwargs: tp.Any
+    ) -> tp.Callable[..., tp.Any]:
```
Comment (boehmseb) [inline varats/varats/tools/tool_util.py:78]: My mistake. Either, we allow functions with return types to be wrapped, then we need to add a `return` in line 80, or we do not allow returns, then we need to do this change:
```suggestion
    ) -> tp.Callable[..., None]:
```

**Review by boehmseb — APPROVED** [review]

**Pull request merged**